src/applications/PrepareDataset.py

```diff
@@ -202,262 +231,217 @@ def copyFilesToCorrectLocation(interimOutputDir, finalSubjectOutputDir, subjectI
     return runBratsPipeline, expected_outputs
 
 
-def main():
-    copyrightMessage = (
-        "Contact: admin@fets.ai\n\n"
-        + "This program is NOT FDA/CE approved and NOT intended for clinical use.\nCopyright (c) "
-        + str(date.today().year)
-        + " University of Pennsylvania. All rights reserved."
-    )
-    parser = argparse.ArgumentParser(
-        prog="PrepareDataset",
-        formatter_class=argparse.RawTextHelpFormatter,
-        description="This application calls the BraTSPipeline for all input images and stores the final and intermediate files separately.\n\n"
-        + copyrightMessage,
-    )
-    parser.add_argument(
-        "-inputCSV",
-        type=str,
-        help="The absolute path of the input CSV file containing the list of subjects and their corresponding images",
-        required=True,
-    )
-    parser.add_argument(
-        "-outputDir",
-        type=str,
-        help="The output dir to write the results",
-        required=True,
-    )
-
-    args = parser.parse_args()
-
-    assert os.path.exists(args.inputCSV), "Input CSV file not found"
-
-    outputDir_qc = posixpath.join(args.outputDir, "DataForQC")
-    outputDir_final = posixpath.join(args.outputDir, "DataForFeTS")
-
-    for dir_to_create in [args.outputDir, outputDir_qc, outputDir_final]:
-        Path(dir_to_create).mkdir(parents=True, exist_ok=True)
-
-    bratsPipeline_exe = os.path.join(
-        Path(__file__).parent.resolve(), "BraTSPipeline"
-    )
-    if platform.system().lower() == "windows":
-        bratsPipeline_exe += ".exe"
-
-    assert os.path.exists(
-        bratsPipeline_exe
-    ), "BraTS Pipeline executable not found, please contact admin@fets.ai for help."
-
-    # only parse the headers here
-    parsed_headers = parse_csv_header(args.inputCSV)
+class Preparator:
+    def __init__(self, input_csv: str, output_dir: str):
+        self.input_csv = input_csv
+        self.input_dir = str(Path(input_csv).parent)
+        self.output_dir = os.path.normpath(output_dir)
+        self.interim_output_dir = posixpath.join(self.output_dir, "DataForQC")
+        self.final_output_dir = posixpath.join(self.output_dir, "DataForFeTS")
+        self.subjects_file = posixpath.join(self.final_output_dir, "processed_data.csv")
+        self.neg_subjects_file = posixpath.join(
+            self.final_output_dir, "QC_subjects_with_negative_intensities.csv"
+        )
+        self.failing_subjects_file = posixpath.join(
+            self.final_output_dir, "QC_subjects_with_bratspipeline_error.csv"
+        )
+        self.__init_out_dfs()
+        self.stdout_log = posixpath.join(self.output_dir, "preparedataset_stdout.txt")
+        self.stderr_log = posixpath.join(self.output_dir, "preparedataset_stderr.txt")
+        self.dicom_tag_information_to_write_collab = {}
+        self.dicom_tag_information_to_write_anon = {}
+        self.brats_pipeline_exe = "BraTSPipeline"
```

Comment:
Is there any way we can allow running `BraTSPipeline` from another path? Inside the MLCube I'm running scripts outside the Front-End path, and I'm moving the binaries to `/usr/bin` so that I can call them directly by name. If not I'll see how I can use this suggested logic inside the MLCube